pkg/deployment/resources/pdbs.go

```diff
@@ -0,0 +1,105 @@
+)
+
+func min(a int, b int) int {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+// EnsurePDBs ensures Pod Distruption Bugets for different server groups in Cluster mode
+func (r *Resources) EnsurePDBs() error {
+
+	// Only in Cluster and Production Mode
+	spec := r.context.GetSpec()
+	if spec.IsProduction() && spec.GetMode().IsCluster() {
+
+		minAgents := spec.GetServerGroupSpec(api.ServerGroupAgents).GetCount() - 1
+		minDBServers := spec.GetServerGroupSpec(api.ServerGroupDBServers).GetCount() - 1
+		minCoordinators := min(spec.GetServerGroupSpec(api.ServerGroupCoordinators).GetCount()-1, 2)
+
+		if err := r.ensurePDBForGroup(api.ServerGroupAgents, minAgents); err != nil {
+			return err
+		}
+		if err := r.ensurePDBForGroup(api.ServerGroupDBServers, minDBServers); err != nil {
+			return err
+		}
+		if err := r.ensurePDBForGroup(api.ServerGroupCoordinators, minCoordinators); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func pdbNameForGroup(depl string, group api.ServerGroup) string {
+	return fmt.Sprintf("%s-%s-pdb", depl, group.AsRole())
+}
+
+func newPDB(minAvail int, deplname string, group api.ServerGroup, owner metav1.OwnerReference) *policyv1beta1.PodDisruptionBudget {
+	return &policyv1beta1.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            pdbNameForGroup(deplname, group),
+			OwnerReferences: []metav1.OwnerReference{owner},
+		},
+		Spec: policyv1beta1.PodDisruptionBudgetSpec{
+			MinAvailable: newFromInt(minAvail),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: k8sutil.LabelsForDeployment(deplname, group.AsRole()),
+			},
+		},
+	}
+}
+
+// ensurePDBForGroup ensure pdb for a specific server group
+func (r *Resources) ensurePDBForGroup(group api.ServerGroup, wantedMinAvail int) error {
+	deplname := r.context.GetAPIObject().GetName()
+	pdbname := pdbNameForGroup(deplname, group)
+	pdbcli := r.context.GetKubeCli().Policy().PodDisruptionBudgets(r.context.GetNamespace())
+	log := r.log.With().Str("group", group.AsRole()).Logger()
+
+	pdb, err := pdbcli.Get(pdbname, metav1.GetOptions{})
+	if k8sutil.IsNotFound(err) {
+		// No PDB found - create new
+		pdb := newPDB(wantedMinAvail, deplname, group, r.context.GetAPIObject().AsOwner())
+		log.Debug().Msg("Creating new PDB")
+		if _, err := pdbcli.Create(pdb); err != nil {
+			log.Error().Err(err).Msg("failed to create PDB")
+			return maskAny(err)
+		}
+		return nil
+	} else if err == nil {
+		// PDB is there
+		if pdb.Spec.MinAvailable.IntValue() != wantedMinAvail {
+			log.Debug().Int("wanted-min-avail", wantedMinAvail).
+				Int("current-min-avail", pdb.Spec.MinAvailable.IntValue()).
+				Msg("Updating PDB")
+			pdb.Spec.MinAvailable = newFromInt(wantedMinAvail)
+
+			// Update the PDB
+			_, err := pdbcli.Update(pdb)
+			if err != nil {
+				log.Error().Err(err).Msg("PDB update failed")
+				maskAny(err)
+			}
+			return nil
+		}
+
+	}
+
+	return maskAny(err)
+}
+
+func newFromInt(v int) *intstr.IntOrString {
```

Comment:
Replace (or use) with `intstr.FromInt(...)` ?

Reply:
`intstr.FromInt` returns by value, I need by pointer. I could do 
```
ret = &intstr.IntOrString
*ret = intstr.FromInt(v)
```

Reply:
That is better than building your own helper that tweaks with structs